### ISSUE TITLE
Basic functionality of set in not working

### ISSUE THREAD

**Issue by piyushchauhan** [bluefireteam/ordered_set#4]
@luanpotter Basic set functionality is not working in Dart VM version: 2.8.4 (stable)
```
OrderedSet<int> items = OrderedSet();
items.add(2);
items.add(1);
items.add(2);
print(items.toList());
```
Result:    `[1, 2, 2]`
Expected: `[1, 2]`

**Comment by luanpotter** [comment]
> Unlike Dart's SplayTreeSet, it allows for several different elements with the same priority to be added.

So it's intended that elements with same priority are allowed. Not sure if we should allow or not identical elements, up to debate.

But there is indeed a weird issue if you are seeing `3` there haha. Should be [1, 2, 2]

**Comment by luanpotter** [comment]
Added a test for it here, seem to be working as expected: https://github.com/luanpotter/ordered_set/pull/7

**Comment by jbx1** [comment]
> So it's intended that elements with same priority are allowed. Not sure if we should allow or not identical elements, up to debate.

If identical elements are allowed, then it is not a Set any more by definition, it is a priority queue.

**Comment by spydon** [comment]
> > So it's intended that elements with same priority are allowed. Not sure if we should allow or not identical elements, up to debate.
> 
> If identical elements are allowed, then it is not a Set any more by definition, it is a priority queue. 
> 
> 

It is a set since the objects are unique, not the priority of the objects.

**Comment by jbx1** [comment]
@spydon That is precisely my point. That previous statement I quoted means that it is not clear whether uniqueness is guaranteed or not. If it is a set, then the objects must be unique according some definition of equality. If this is not guaranteed (the "up to debate" thing) then it should not be called a set. Their priority ordering has nothing to do with it (i never mentioned it in fact).

**Comment by spydon** [comment]
> @spydon That is precisely my point. That previous statement I quoted means that it is not clear whether uniqueness is guaranteed or not. If it is a set, then the objects must be unique according some definition of equality. If this is not guaranteed (the "up to debate" thing) then it should not be called a set.

Aah missed parts of the thread, you are correct.

**Comment by spydon** [comment]
This was fixed in #19.
